apps/emqx_rule_engine/src/emqx_rule_utils.erl

```diff
@@ -121,11 +121,11 @@ proc_cmd(Tokens, Data, Opts) ->
     [proc_tmpl(Tks, Data, Opts) || Tks <- Tokens].
 
 %% preprocess SQL with place holders
--spec(preproc_sql(Sql::binary()) -> {prepare_statement(), prepare_params()}).
+-spec(preproc_sql(Sql::binary()) -> {prepare_statement(), [prepare_params()]}).
```

Comment:
These specs in the app `emqx_rule_engine` are a little bit complex and I will fix them in another PR.

You could keep them unchanged for now.

Reply:
@ayo-dele maybe add a suppression for now.